src/esp/gfx/Drawable.h

```diff
@@ -53,14 +59,25 @@ class Drawable : public Magnum::SceneGraph::Drawable3D {
    * @param transformationMatrix  Transformation relative to camera.
    * @param camera                Camera to draw from.
    *
-   * Each derived drawable class needs to implement this draw() function. It's
-   * nothing more than drawing itself with its group's shader.
+   * Each derived drawable class needs to implement this draw() function.
+   * It's nothing more than drawing itself with its group's shader.
    */
   virtual void draw(const Magnum::Matrix4& transformationMatrix,
                     Magnum::SceneGraph::Camera3D& camera) = 0;
 
+  static uint64_t drawableIdCounter;
+  uint64_t drawableId_;
+
   scene::SceneNode& node_;
   Magnum::GL::Mesh& mesh_;
+
+  /**
+   * Why a friend class?
+   * because DrawableGroup will have to update the state, attachedToGroup
+   * directly, and ONLY this class can do it
+   */
+  friend class DrawableGroup;
+  bool attachedToGroup_ = false;
```

Comment:
hm actually isn't this state already present in magnum's feature class? there's [group()](https://doc.magnum.graphics/magnum/classMagnum_1_1SceneGraph_1_1AbstractGroupedFeature.html#a670e9a5ed8e81ed9ea5b819e3af01ee6) (or `drawable()`), which returns a pointer -- if it's null, the feature is not in any group

Reply:
We need it. we need to make sure it is attaching to our DrawableGroup (sub-class of a magnum drawable group), not the magnum base drawable group.